### PR TITLE
P4Fmt reference checker

### PR DESCRIPTION
**Pull request by snapdgn**
reference checker for `p4fmt` formatter

### CODE REVIEW

**Comment by qobilidop** [comment]
Not an immediate requirement, but I think it would be great to integrate this checker with GitHub CIs eventually. For now, adding a section in the README describing its usage might be good enough.

File: backends/p4fmt/p4fmt.h [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#ifndef BACKENDS_P4FMT_P4FMT_H_
+#define BACKENDS_P4FMT_P4FMT_H_
+
+#include <filesystem>
+#include <sstream>
+
+std::stringstream getFormattedOutput(std::filesystem::path inputFile);
```
Comment (fruffy) [inline backends/p4fmt/p4fmt.h:7]: Please add a triple slash comment describing this function here.

File: backends/p4fmt/p4fmt.cpp [hunk 2]
```diff
@@ -1,51 +1,35 @@
...
-            return EXIT_FAILURE;
-        }
-    }
+    std::stringstream formattedOutput;
 
     const IR::P4Program *program = P4::parseP4File(options);
-
     if (program == nullptr && ::errorCount() != 0) {
-        return EXIT_FAILURE;
+        std::cerr << "Error: Failed to parse P4 file." << std::endl;
```
Comment (fruffy) [inline backends/p4fmt/p4fmt.cpp:21]: Use `::error("..")` here

**Review by fruffy — CHANGES_REQUESTED** [review]

File: backends/p4fmt/p4fmt.cpp [hunk 3]
```diff
@@ -1,51 +1,36 @@
...
 #include "frontends/p4/toP4/toP4.h"
 #include "ir/ir.h"
 #include "lib/compile_context.h"
-#include "lib/cstring.h"
 #include "lib/error.h"
-#include "lib/nullstream.h"
 #include "options.h"
 
-int main(int argc, char *const argv[]) {
+/// Formats a P4 program from the input file, returns formatted output
```
Comment (fruffy) [inline backends/p4fmt/p4fmt.cpp:11]: This should be in the header file.

**Review by fruffy — APPROVED** [review]

**Pull request merged**